contracts/strategy/rysk/RyskStrategy.sol

```diff
@@ -0,0 +1,265 @@
+    }
+
+    //
+    // IStrategy
+    //
+
+    /**
+     * Rysk strategy is asynchronous meaning it doesn't support immediate withdrawals.
+     *
+     * @return false always
+     */
+    function isSync() external pure override(IStrategy) returns (bool) {
+        return false;
+    }
+
+    /// @inheritdoc IStrategy
+    function hasAssets()
+        external
+        view
+        virtual
+        override(IStrategy)
+        returns (bool)
+    {
+        // TODO: refactor & test
+        if (
+            pendingWithdrawal.shares != 0 ||
+            ryskLqPool.balanceOf(address(this)) != 0
+        ) return true;
+
+        return ryskLqPool.depositReceipts(address(this)).epoch != 0;
+    }
+
+    /// @inheritdoc IStrategy
+
+    // TODO: has to redeem before calculating assets!!!
+    /// @notice this also includes shares from the pending withdrawal
+    function investedAssets()
+        public
+        view
+        virtual
+        override(IStrategy)
+        returns (uint256)
+    {
+        IRyskLiquidityPool.DepositReceipt memory depositReceipt = ryskLqPool
+            .depositReceipts(address(this));
+        uint256 currentPricePerShare = _getWithdrawalPricePerShare();
+
+        // shares yet to be minted for pending deposits
+        uint256 pendingSharesAmount = 0;
+        if (depositReceipt.epoch != 0) {
+            uint256 depositPricePerShare = ryskLqPool.depositEpochPricePerShare(
+                depositReceipt.epoch
+            );
+
+            pendingSharesAmount =
+                (((depositReceipt.amount * SHARES_CONVERSION_FACTOR) /
+                    depositPricePerShare) * currentPricePerShare) /
+                SHARES_CONVERSION_FACTOR;
+        }
+
+        // shares marked by initiated withdrawal (now owned by the pool itself)
+        uint256 withdrawalSharesAmount = (pendingWithdrawal.shares *
+            ryskLqPool.withdrawalEpochPricePerShare(pendingWithdrawal.epoch)) /
+            SHARES_CONVERSION_FACTOR;
+
+        // entitled = redeemed + unredeemed
+        uint256 entitledSharesAmount = ((depositReceipt.unredeemedShares +
+            ryskLqPool.balanceOf(address(this))) * currentPricePerShare) /
+            SHARES_CONVERSION_FACTOR;
+
+        return
+            pendingSharesAmount + withdrawalSharesAmount + entitledSharesAmount;
+    }
+
+    /// @inheritdoc IStrategy
+    function invest() external virtual override(IStrategy) onlyManager {
+        uint256 balance = underlying.balanceOf(address(this));
+
+        if (balance == 0) revert StrategyNoUnderlying();
+
+        emit StrategyInvested(balance);
+        ryskLqPool.deposit(balance);
+    }
+
+    /// @inheritdoc IStrategy
+    function withdrawToVault(uint256 _amount)
+        external
+        virtual
+        override(IStrategy)
+        onlyManager
+    {
+        if (_amount == 0) revert StrategyAmountZero();
+        if (_amount > investedAssets()) revert StrategyNotEnoughShares();
```

Comment:
This error is a bit weird. I know it's the same we use for Liquity, but the wording is incorrect. The "shares" in this error comes from the Yearn shares. We should consider renaming this for something like `StrategyCannotWithdrawMoreThanTheAvailableAmount`.

Reply:
I've changed this logic now since it was not consistent with how liquidity pool, take another look